earth_enterprise/src/scons/getversion.py

```diff
@@ -222,17 +242,30 @@ def set_long(self, value):
 
         self.long_version_string = value
 
+    def get_warning_message(self):
+        """Returns None, or a string describing known issues."""
+
+        return None if _IsGitDescribeFirstParentSupported() else '''\
+WARNING: Git version 1.8.4 or later is required to correctly determine the Open GEE version being built.
+The Open GEE version is calculated from tags using the "git describe" command.
+The "--first-parent" parameter introduced in Git 1.8.4 allows proper version calcuation on all branches.
+Without the --first-parent parameter, the version calculated may be incorrect, depending on which branch is being built.
+For information on upgrading Git, see:
+https://github.com/google/earthenterprise/wiki/Frequently-Asked-Questions-(FAQs)#how-do-i-upgrade-git-to-the-recommended-version-for-building-google-earth-enterprise\
+'''
+
 
 # Exported variable for use by other modules:
 open_gee_version = OpenGeeVersion()
 
-
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument("-l", "--long", action="store_true", help="Output long format of version string")
     args = parser.parse_args()
 
     print open_gee_version.get_long() if args.long else open_gee_version.get_short()
+    if open_gee_version.get_warning_message():
+        print >> sys.stderr, open_gee_version.get_warning_message()
```

Comment:
calling `get_warning_message` twice seems undesirable to me, because there are system calls and a several-levels-deep call tree involved under the hood. And calling the same function on two lines in a row always looks odd to me.

Perhaps store the result and check it? It would  read pretty well:
```
    warning_message = open_gee_version.get_warning_message()
    if warning_message is not None:
        print >> sys.stderr, warning_message
```

Reply:
Good callout -- fixed.